### ISSUE TITLE
1.6 update issue: check constraint "file_base_check" is violated

### ISSUE THREAD

**Issue by freelock** [ledgersmb/LedgerSMB#3629]
When updating from 1.5 to 1.6rc1, setup.pl database upgrade fails with:

Error!

Error applying upgrade script ./sql/changes/1.5/abstract_tables.sql: ERROR: check constraint "file_base_check" is violated by some row at /srv/ledgersmb/bin/../lib/LedgerSMB/Database/Change.pm line 227.

dbversion: , company: 1.6.0-rc1

The error occurs when a check no inherit constraint is added to the file_base table.

On my 1.5 database, file_base shows these inherited tables:

```
freelockco=# \d+ file_base
  ...
Indexes:
    "file_base_pkey" PRIMARY KEY, btree (ref_key, file_name, file_class)
    "file_base_id_key" UNIQUE CONSTRAINT, btree (id)
Foreign-key constraints:
    "file_base_file_class_fkey" FOREIGN KEY (file_class) REFERENCES file_class(i
d)
    "file_base_mime_type_id_fkey" FOREIGN KEY (mime_type_id) REFERENCES mime_typ
e(id)
    "file_base_uploaded_by_fkey" FOREIGN KEY (uploaded_by) REFERENCES entity(id)
Child tables: file_eca,
              file_entity,
              file_incoming,
              file_internal,
              file_order,
              file_part,
              file_transaction

**Comment by freelock** [comment]
This happens with databases that had files originally uploaded in LSMB 1.3, and have been updated to 1.6.

The file_base table has 7 child tables. This check constraint verifies that all rows in the file_base table are in a child table -- however in 1.3 they appear to be in the base table and a child table.

A workaround appears to be deleting the extra row in file_base, using the following:

> DELETE FROM ONLY file_base;

**Comment by ehuelsmann** [comment]
A schema migration check has been offered to @freelock  for testing.

**Comment by ehuelsmann** [comment]
Merged to master in dd6934f7

**Comment by ehuelsmann** [comment]
Merged to 1.6 in 6aa30b1f
